File: scripts/nickname-map.js
/**
 * Resolve session nicknames to real player names.
 * Mattheus, Shelley, and Emily each use multiple alter egos per session.
 */
const NICKNAME_TO_PLAYER = {
  // Mattheus
  SHREKK: 'Mattheus',
  MATE: 'Mattheus',
  MA: 'Mattheus',
  MAP: 'Mattheus',
  MAE: 'Mattheus',
  MEH: 'Mattheus',
  // Shelley
  MAFUS: 'Shelley',
  SHRIMP: 'Shelley',
  SHEL: 'Shelley',
  SHE: 'Shelley',
  WIMBY: 'Shelley',
  // Emily
  EMLIE: 'Emily',
  EH: 'Emily',
};

function resolvePlayerFromNickname(nickname) {
  const key = String(nickname || '').trim().toUpperCase();
  return NICKNAME_TO_PLAYER[key] || null;
}

module.exports = { NICKNAME_TO_PLAYER, resolvePlayerFromNickname };

File: tools/nickname_map.py
"""Alley nickname → real player name (mirrors js/nickname-map.js)."""

NICKNAME_TO_PLAYER: dict[str, str] = {
    "SHREKK": "Mattheus",
    "MATE": "Mattheus",
    "MA": "Mattheus",
    "MAP": "Mattheus",
    "MAE": "Mattheus",
    "MEH": "Mattheus",
    "MAFUS": "Shelley",
    "SHRIMP": "Shelley",
    "SHEL": "Shelley",
    "SHE": "Shelley",
    "WIMBY": "Shelley",
    "EMLIE": "Emily",
    "EH": "Emily",
}


def resolve_player_from_nickname(nickname: str) -> str | None:
    key = (nickname or "").strip().upper()
    return NICKNAME_TO_PLAYER.get(key)
